packages/elements/list/src/withList.spec.tsx

```diff
@@ -303,4 +314,182 @@ describe('withList', () => {
+        </editor>
+      ) as any) as Editor;
+
+      const expected = ((
+        <editor>
+          <hul>
+            <hli>
+              <hlic>
+                level 1
+                <cursor />
+                level 2
+              </hlic>
+            </hli>
+          </hul>
+        </editor>
+      ) as any) as Editor;
+
+      testDeleteForward(input, expected);
+    });
+
+    it('should merge next li and shift one level up', () => {
+      const input = ((
+        <editor>
+          <hul>
+            <hli>
+              <hlic>level 1</hlic>
+              <hul>
+                <hli>
+                  <hlic>
+                    level 2<cursor />
+                  </hlic>
+                </hli>
+              </hul>
+            </hli>
+            <hli>
+              <hlic>level 3</hlic>
+              <hul>
+                <hli>
+                  <hlic>level 4</hlic>
+                </hli>
+              </hul>
+            </hli>
+          </hul>
+        </editor>
+      ) as any) as Editor;
+
+      const expected = ((
+        <editor>
+          <hul>
+            <hli>
+              <hlic>level 1</hlic>
+              <hul>
+                <hli>
+                  <hlic>level 2level 3</hlic>
+                </hli>
+                <hli>
+                  <hlic>level 4</hlic>
+                </hli>
+              </hul>
+            </hli>
+          </hul>
+        </editor>
+      ) as any) as Editor;
+
+      testDeleteForward(input, expected);
+    });
+
+    it('should shift all nested lists one level up', () => {
+      const input = ((
+        <editor>
+          <hul>
+            <hli>
+              <hlic>
+                level 1<cursor />
+              </hlic>
+              <hul>
+                <hli>
+                  <hlic>level 2</hlic>
+                  <hul>
+                    <hli>
+                      <hlic>level 3</hlic>
+                      <hul>
+                        <hli>
+                          <hlic>level 4</hlic>
+                        </hli>
+                      </hul>
+                    </hli>
+                    <hli>
+                      <hlic>level 5</hlic>
+                    </hli>
+                  </hul>
+                </hli>
+              </hul>
+            </hli>
+          </hul>
+        </editor>
+      ) as any) as Editor;
+
+      const expected = ((
+        <editor>
+          <hul>
+            <hli>
+              <hlic>
+                level 1
+                <cursor />
+                level 2
+              </hlic>
+              <hul>
+                <hli>
+                  <hlic>level 3</hlic>
+                  <hul>
+                    <hli>
+                      <hlic>level 4</hlic>
+                    </hli>
+                  </hul>
+                </hli>
+                <hli>
+                  <hlic>level 5</hlic>
+                </hli>
+              </hul>
+            </hli>
+          </hul>
```

Comment:
```suggestion
          </hul>
            <hli>
              <hlic>level 1</hlic>
            </hli>
          </hul>
```

I suggest adding an li after the nested list to make sure nothing strange happens to it now or in the future.